csm/conf/setup.py

```diff
@@ -160,8 +185,13 @@ def _config_user(self, reset=False):
         """
         if not reset:
             if not self._is_user_exist():
-                Setup._run_cmd("useradd -d "+const.CSM_USER_HOME+" -p "+self._password+" "+ self._user)
+                _password = self._fetch_csm_user_password(decrypt=True)
+                if not _password:
+                    Log.error("CSM Password Not Recieved from provisioner.")
+                    raise CsmSetupError("CSM Password Not Set by Provisioner.")
+                Setup._run_cmd("useradd -d "+const.CSM_USER_HOME+" -p "+_password+" "+ self._user)
                 Setup._run_cmd("usermod -aG wheel " + self._user)
+                Setup._run_cmd("usermod -s /sbin/nologin " + self._user)
```

Comment:
You can remove this, this was part of alternate solution.

Reply:
"nologin" is required. Do not remove it.